remotespark/livyclientlib/livysession.py

```diff
@@ -90,11 +89,14 @@ def create_sql_context(self):
         self.ipython_display.writeln("Creating HiveContext as 'sqlContext'")
         command = self._get_sql_context_creation_command()
         try:
-            command.execute(self)
+            (success, out) = command.execute(self)
         except LivyClientTimeoutError:
             raise LivyClientTimeoutError("Failed to create the SqlContext in time. Timed out after {} seconds."
                                          .format(self._wait_for_idle_timeout_seconds))
-        self.created_sql_context = True
+        if success:
```

Comment:
I'm advising Mohamed to add, at the very least, to add a unit test to the magics class to ensure it logs a fatal error when adding the session fails (if that unit test doesn't already exist).  He can also add a unit test to `SparkController` to verify it raises an exception if `session.start()` throws which would be very easy.  That would be enough verification.

Another thing you can do for some further assurance that this exception will not be spontaneously caught is to use a different exception than `ValueError`, like a custom exception `FailedToCreateSqlContextError`.